packages/forms/src/patterns/attachment/response.ts

```diff
@@ -0,0 +1,62 @@
+import { z } from 'zod';
+import { ParseUserInput } from '../../pattern.js';
+import { safeZodParseToFormError } from '../../util/zod.js';
+import { type AttachmentPattern } from './config.js';
+
+const createSchema = (data: AttachmentPattern['data']) => {
+  // Define the maximum file size in bytes
+  const maxFileSizeBytes = data.maxFileSizeMB * 1024 * 1024;
+
+  // Define the schema for a single file in the FileList
+  const fileSchema = z
+    .instanceof(File)
+    .refine((file: File) => data.allowedFileTypes.includes(file.type), {
+      message: `Invalid file type`,
+    })
+    .refine((file: File) => file.size <= maxFileSizeBytes, {
+      message: `Each file must be smaller than ${data.maxFileSizeMB} MB`,
+    });
+
+  // Define the schema for the FileList as a whole
+  return z
+    .custom<FileList>(
```

Comment:
This seems to only work on the Spotlight app, due to FileList not being defined on node.js. We also need to make sure that whatever we store as session data is JSON-serializable.

We may want to add a new high-level concept to the session object, to deal with references to blobs of data, rather than embedding them, to deal with larger objects. Let's discuss this more this week.

Reply:
Architecturally, this requires a bit more discussion. In order to handle the file input type via a form submit event with HTTP POST, the form requires [`enctype="multipart/form-data"`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/enctype) and additional consideration server-side. 

If this uses body-parser either in Express or as a dependency with the local Astro server, which would be a logical choice for a JSON-only expectation, there would need to be some other type of handling for the upload as [body-parser's README](https://github.com/expressjs/body-parser) mentions that it doesn't handle multipart encodings. Changing this would have an effect on other parts of the form.

There are some short-term solutions that we could revisit as part of the form-filler user persona and epic. One option would be to just do this all with JS. We could encode the files and post to a separate endpoint when someone adds a form to the component. This would rely on JS. 

Another option that might be worth investigating is to have this component live inside of an `<iframe>` that has a `<form>` with the correct encoding for attachments on it that hits a different endpoint.

Reply:
Are there downsides you'd see to setting the enctype? Server-side, we unpack FormData from the body [here](https://github.com/GSA-TTS/forms/blob/main/packages/server/src/pages/forms/%5Bid%5D.astro#L42-L46).

Reply:
No, there aren't any downsides to it. I actually had the enctype on there during local development, but when I was logging it server-side, the File object was passed in as a string in the validation function. Now that I see what you linked to, I can see why that was happening. 

I'll see if it's a low LOE thing to wrap up in conjunction with this ticket, and if so, I'll do it. If not, we'll write it as a ticket to address when we work on the form filler epic and a later PR.

Reply:
This now works with the multipart enctype. Files are getting passed in as an array, and ultimately will get transformed to a base64 encoded string. There's a placeholder transform in the validation logic so we can mull it over a little more and discuss when we get to the form filler epic.